openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1432 @@
+      timer.restart("Changing asymmetric background value");
+#endif
+      changeAsymmetricLevelSetBackground(mGrid1->tree(), e.max(), e.min());//multi-threaded
+#ifdef BENCHMARK_FAST_SWEEPING
+      timer.stop();
+#endif
+    }
+}// FastSweeping::sweep
+
+/// Private class of FastSweeping to quickly compute the extrema
+/// values of the active voxels in the leaf nodes. Several orders
+/// of magnitude faster than tools::extrema!
+template <typename GridT>
+struct FastSweeping<GridT>::MinMaxKernel
+{
+    using LeafMgr = tree::LeafManager<const TreeT>;
+    using LeafRange = typename LeafMgr::LeafRange;
+    MinMaxKernel() : mMin(std::numeric_limits<ValueT>::max()), mMax(-mMin) {}
+    MinMaxKernel(MinMaxKernel& other, tbb::split) : mMin(other.mMin), mMax(other.mMax) {}
+    math::MinMax<ValueT> run(const GridT &grid)
+    {
+        LeafMgr mgr(grid.tree());// super fast
+        tbb::parallel_reduce(mgr.leafRange(), *this);
+        return math::MinMax<ValueT>(mMin, mMax);
+    }
+    void operator()(const LeafRange& r)
+    {
+        for (auto leafIter = r.begin(); leafIter; ++leafIter) {
+            for (auto voxelIter = leafIter->beginValueOn(); voxelIter; ++voxelIter) {
+                const ValueT v = *voxelIter;
+                if (v < mMin) mMin = v;
+                if (v > mMax) mMax = v;
+            }
+        }
+    }
+    void join(const MinMaxKernel& other)
+    {
+        if (other.mMin < mMin) mMin = other.mMin;
+        if (other.mMax > mMax) mMax = other.mMax;
+    }
+    ValueT mMin, mMax;
+};//MinMaxKernel
+
+////////////////////////////////////////////////////////////////////////////////
+
+/// Private class of FastSweeping to perform multi-threaded initialization
+template <typename GridT>
+struct FastSweeping<GridT>::DilateKernel
+{
+    using LeafRange = typename tree::LeafManager<TreeT>::LeafRange;
+    using BufferT = typename FastSweeping::CoordArrayT::ValueBuffer;//util::PagedArray<Coord, 10>::ValueBuffer;
+    using PoolT = tbb::enumerable_thread_specific<BufferT>;
+    DilateKernel(FastSweeping &parent)
+        : mParent(&parent), mPool(nullptr), mBackground(parent.mGrid1->background())
+    {
+    }
+    DilateKernel(const DilateKernel &parent) = default;// for tbb::parallel_for
+    DilateKernel& operator=(const DilateKernel&) = delete;
+
+    void run(int dilation, NearestNeighbors nn)
+    {
+#ifdef BENCHMARK_FAST_SWEEPING
+        util::CpuTimer timer("Construct LeafManager");
+#endif
+        tree::LeafManager<TreeT> mgr(mParent->mGrid1->tree());// super fast
+
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Changing background value");
+#endif
+        static const ValueT Unknown = std::numeric_limits<ValueT>::max();
+        changeLevelSetBackground(mgr, Unknown);//multi-threaded
+
+ #ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Dilating and updating mgr (parallel)");
+        //timer.restart("Dilating and updating mgr (serial)");
+#endif
+
+        const int delta = 5;
```

Comment:
What does this delta refer too?

Reply:
the value five :-D This is the sub-division that seemed to result in the best performance

Reply:
Ah okay understood! If it remains a comment would be good, but it sounds like we agreed on replacing the below with a single call to `dilateActiveValues`